### PR TITLE
Dfa pyrun

### PR DESCRIPTION
**Pull request by matthiasto**

### CODE REVIEW

File: avaframe/com1DFAPy/com1DFA.py [hunk 1]
```diff
@@ -55,6 +55,177 @@
...
+    cfgGen = cfg['GENERAL']
+    modName = 'com1DFAPy'
+
+    # Log current avalanche directory
+    log.debug('Your current avalanche name: %s' % avaDir)
+
+    # Create output and work directories
+    workDir, outDir = inDirs.initialiseRunDirs(avaDir, modName)
+
+    # Load input data
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:86]: Why was this moved here instead of using the initializeSimulations function?

File: avaframe/com1DFAPy/com1DFA.py [hunk 2]
```diff
@@ -55,6 +55,177 @@
...
+        for k in range(len(relDict['d0'])):
+            if relDict['d0'][k] == 'None':
+                relDict['d0'][k] = '1.0'
+        log.info('Release area scenario: %s - perform simulations' % (relName))
+        if flagEntRes:
+            log.info('Entrainment area: %s and resistance area: %s' % (entrainmentArea, resistanceArea))
+
+        if flagEntRes:
+            # Initialise CreateSimulations cint file and set parameters
+            cuSim = [simName + '_null_dfa', simName + '_entres_dfa']
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:144]: Twice the same flag...

File: avaframe/com1DFAPy/com1DFA.py [hunk 3]
```diff
@@ -55,6 +55,177 @@
...
+            cuSim = [simName + '_null_dfa']
+        if flagOnlyEntrRes:
+            cuSim = [simName + '_entres_dfa']
+
+        for sim in cuSim:
+            if flagEntRes:
+                log.debug('One simulation is performed using entrainment and \
+                           one standard simulation without')
+            else:
+                log.debug('Standard simulation is performed without entrainment and resistance')
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:156]: Again, same flag as above with no additional info

File: avaframe/com1DFAPy/com1DFA.py [hunk 4]
```diff
@@ -55,6 +55,177 @@
...
+            logName = sim + '_' + cfgGen['mu']
+            inDirPart = ''
+            if cfgGen.getboolean('initialiseParticlesFromFile'):
+                partDirName = logName
+                inDirPart = os.path.join(cfgGen['avalancheDir'], 'Outputs', 'com1DFA', 'particles', partDirName)
+            if cfgGen.getboolean('benchmarks'):
+                avaName = os.path.basename(cfgGen['avalancheDir'])
+                inDirPart = os.path.join('..', 'benchmarks', avaName, 'particles', partDirName)
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:164]: So if benchmark is run, particles are always used? Is this really what is wanted? Or is this for development purposes for now?

File: avaframe/com1DFAPy/com1DFACfg.ini [hunk 5]
```diff
@@ -97,6 +97,8 @@ initialiseParticlesFromFile = False
 flagEnt = True
 # True if resistance areas shall be included
 flagRes = True
+# True if only the entrainment resistance simulation should be done
+flagOnlyEntrRes = False
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFACfg.ini:101]: See discussion on matrix about the usage of flagSomething...

File: avaframe/runStandardTestsDFA.py [hunk 6]
```diff
@@ -27,15 +27,19 @@
...
+                 # 'data/avaMal',
+                 # 'data/avaWog',
+                 # 'data/avaBowl',
+                 # 'data/avaFlatPlane',
+                 # 'data/avaHelix',
+                 # 'data/avaHelixChannel',
+                 # 'data/avaHockey',
+                 # 'data/avaHockeySmoothChannel',
+                 # 'data/avaHockeySmoothSmall',
+                 # 'data/avaInclinedPlane'
```
Comment (fso42) [inline avaframe/runStandardTestsDFA.py:42]: Is this intended to be commented out?

File: avaframe/com1DFAPy/com1DFACfg.ini (hunk 5 above)
Comment (fso42) [reply avaframe/com1DFAPy/com1DFACfg.ini]: see issue #86

File: avaframe/com1DFAPy/com1DFA.py [hunk 7]
```diff
@@ -55,6 +53,187 @@
...
+        # find out which simulations to perform
+        relName, cuSim, relDict, BadName = getSimulation(cfg['FLAGS'], rel)
+        for sim in cuSim:
+            if '_entres_' in sim:
+                if entLine:
+                    log.info('Simulation is performed using entrainment: %s' % (entLine['Name']))
+                if resLine:
+                    log.info('Simulation is performed using resistance: %s' % (resLine['Name']))
+            else:
+                log.info('Standard simulation is performed without entrainment and resistance')
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:111]: This seems a bit redundant, since it is just printouts... I'd completely drop it...

File: avaframe/com1DFAPy/com1DFA.py [hunk 8]
```diff
@@ -55,6 +53,187 @@
...
+            reportDictList.append(reportST)
+
+            # Count total number of simulations
+            countRel = countRel + 1
+    log.debug('Avalanche Simulations performed')
+
+    return Particles, Fields, Tsave, dem, reportDictList
+
+
+def getSimulation(cfgFlags, rel):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:155]: Please add a docstring

File: avaframe/com1DFAPy/com1DFA.py [hunk 9]
```diff
@@ -55,6 +53,187 @@
...
+    EntRes = cfgFlags.getboolean('EntRes')
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:156]:
```suggestion
    entRes = cfgFlags.getboolean('EntRes')
```

File: avaframe/com1DFAPy/com1DFA.py [hunk 10]
```diff
@@ -55,6 +53,187 @@
...
+    OnlyEntrRes = cfgFlags.getboolean('OnlyEntrRes')
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:157]:
```suggestion
    onlyEntrRes = cfgFlags.getboolean('OnlyEntrRes')
```

File: avaframe/com1DFAPy/com1DFA.py [hunk 11]
```diff
@@ -55,6 +53,187 @@
...
+    # Set release areas and simulation name
+    relName = os.path.splitext(os.path.basename(rel))[0]
+    simName = relName
+    BadName = False
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:161]:
```suggestion
    badName = False
```

File: avaframe/com1DFAPy/com1DFA.py [hunk 12]
```diff
@@ -55,6 +53,187 @@
...
+    # Counter for release area loop
+    countRel = 0
+
+    # Setup simulation dictionaries for report genereation
+    reportDictList = []
+
+    # Loop through release areas
+    for rel in relFiles:
+        # find out which simulations to perform
+        relName, cuSim, relDict, BadName = getSimulation(cfg['FLAGS'], rel)
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:103]:
```suggestion
        relName, cuSim, relDict, badName = getSimulation(cfg['FLAGS'], rel)
```

File: avaframe/com1DFAPy/com1DFA.py [hunk 13]
```diff
@@ -55,6 +53,187 @@
...
+            cuSim = [simName + '_entres_dfa']
+        else:
+            cuSim = [simName + '_null_dfa', simName + '_entres_dfa']
+    else:
+        # Initialise CreateSimulations cint file and set parameters
+        cuSim = [simName + '_null_dfa']
+    return relName, cuSim, relDict, BadName
+
+
+def prepareInputData(demFile, relFiles, entFiles, resFile):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:184]: please add docstring

File: avaframe/com1DFAPy/com1DFA.py [hunk 14]
```diff
@@ -55,6 +53,187 @@
...
+        resistanceArea = os.path.splitext(os.path.basename(resFile))[0]
+        resLine['Name'] = [resistanceArea]
+    else:
+        resLine = None
+        resistanceArea = ''
+
+    return demOri, releaseLine, entLine, resLine, entrainmentArea, resistanceArea
+
+
+def createReportDict(logName, relName, relDict, cfgGen, sim, entrainmentArea, resistanceArea):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:210]: please add docstring

File: avaframe/com1DFAPy/com1DFA.py [hunk 15]
```diff
@@ -55,6 +53,187 @@
...
+
+            # If initialisation from file
+            inDirPart = ''
+            if cfgGen.getboolean('initialiseParticlesFromFile'):
+                if cfgGen['particleFile']:
+                    inDirPart = cfgGen['particleFile']
+                else:
+                    partDirName = logName
+                    inDirPart = os.path.join(cfgGen['avalancheDir'], 'Outputs', 'com1DFA', 'particles', partDirName)
+            cfg['GENERAL']['inDirPart'] = inDirPart
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:122]: Why is this not in the initializeSimulation? Since thats where the particles are initialized... And initializeParticles also has the flag checking in it?

And I think it should only be:
```
if Particles from file:
   Read from file
```
and if the particles produced by com1DFA are wanted, then this is specified in `particleFile` (which can easily done at runtime without the need to touch the ini-File)

File: avaframe/com1DFAPy/com1DFA.py [hunk 16]
```diff
@@ -128,29 +307,10 @@ def setDEMoriginToZero(demOri):
     return dem
 
 
-def initializeSimulation(cfg, relTh):
-    flagDev = cfg['FLAGS'].getboolean('flagDev')
+def initializeSimulation(cfg, demOri, releaseLine, entLine, resLine, logName, relTh):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:310]: please add a docstring

File: avaframe/com1DFAPy/com1DFA.py [hunk 17]
```diff
@@ -445,7 +601,7 @@ def placeParticles(massCell, indx, indy, csz, massPerPart):
     return xpart, ypart, mPart, nPart
 
 
-def initializeMassEnt(dem, flagEntRes, entLine):
+def initializeMassEnt(dem, EntRes, entLine):
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFA.py:604]:
```suggestion
def initializeMassEnt(dem, entRes, entLine):
```

plus below...

File: avaframe/com1DFAPy/com1DFACfg.ini [hunk 18]
```diff
@@ -50,17 +50,17 @@ mindT = 0.05
 # fixed time step [s]
 dt = 0.1
 # save every [s]
-dtSave = 5
+dtSave = 10
 # End time [s]
-Tend = 100
+Tend = 400
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFACfg.ini:55]:
```suggestion
tEnd = 400
```

File: avaframe/com1DFAPy/com1DFACfg.ini [hunk 19]
```diff
@@ -91,12 +91,11 @@ tSteps = 0|50.2|-1
...
 initialiseParticlesFromFile = False
+particleFile =
 
 [FLAGS]
-# True if entrainment areas shall be included
-flagEnt = True
-# True if resistance areas shall be included
-flagRes = True
+# True if only the entrainment resistance simulation should be done
+OnlyEntrRes = False
```
Comment (fso42) [inline avaframe/com1DFAPy/com1DFACfg.ini:98]:
```suggestion
onlyEntrRes = False
```

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com1DFAPy/com1DFA.py (hunk 15 above)
Comment (matthiasto) [reply avaframe/com1DFAPy/com1DFA.py:122]: Not sure I understand exactly what you want...

**Review by fso42 — APPROVED** [review]
